### DISCUSSION TITLE
Getting no-op warning: Can't perform a React state update on an unmounted component.

### DISCUSSION THREAD

**Discussion by HugoSanchez** [rainbow-me/rainbowkit#1220] [Feedback]
Hey there, 

I'm getting this warning when using RainbowKit:

`Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
 `
 
 What's strange enough is that this only happens when selecting Metamask in the modal, but it doesn't happen when selecting Rainbow Wallet for instance. 
 
 I must be doing something wrong but couldn't figure out what it is. Do yo have any clues?
 
 I'm using 
 React 17.0.2
 Nextjs 12.0.3
 @Rainbowkit 0.12.13
 wagmi 0.12.13
 
 I've tried using WalletConnect's web3Modal and the error stops happening. 
 
 Any tips would be much appreciated!
 
 Thanks, 
Hugo

**Comment by DanielSinclair** [comment]
If you're still seeing this issue, please open a new Issue on the repo with a reproducible example and we can investigate.
